server/src/test/java/org/opensearch/index/query/BoolQueryBuilderTests.java

```diff
@@ -456,4 +459,26 @@ public void testMustRewrite() throws IOException {
         IllegalStateException e = expectThrows(IllegalStateException.class, () -> boolQuery.toQuery(context));
         assertEquals("Rewrite first", e.getMessage());
     }
+
+    public void testVisit() {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        boolQueryBuilder.should(new TermQueryBuilder(TEXT_FIELD_NAME, "should"));
+        boolQueryBuilder.must(new TermQueryBuilder(TEXT_FIELD_NAME, "must1"));
+        boolQueryBuilder.must(new TermQueryBuilder(TEXT_FIELD_NAME, "must2")); // Add a second one to confirm that they both get visited
+        boolQueryBuilder.mustNot(new TermQueryBuilder(TEXT_FIELD_NAME, "mustNot"));
+        boolQueryBuilder.filter(new TermQueryBuilder(TEXT_FIELD_NAME, "filter"));
+        List<QueryBuilder> visitedQueries = new ArrayList<>();
+        boolQueryBuilder.visit(createTestVisitor(visitedQueries));
+        assertEquals(5, visitedQueries.size());
```

Comment:
This test is failing on Gradle check, which makes sense: the visitor will be passed the parent `BoolQueryBuilder` and its 5 children, for a total of 6 visited queries.

Reply:
Ack, I have taken care of it and pushed the revision